std/bitmanip.d

```diff
@@ -292,7 +272,208 @@ bool), followed by unsigned types, followed by signed types.
 
 template bitfields(T...)
 {
-    enum { bitfields = createStorageAndFields!T.result }
+    enum { bitfields =  bitfields2CodeImpl!(false, T) }
+}
+
+/**
+Allows creating bit fields inside $(D_PARAM struct)s and $(D_PARAM class)es.
+
+Example:
+
+----
+struct A
+{
+    int a;
+    mixin bitfields!(
+        uint, "x",    2,
+        int,  "y",    3,
+        uint, "z",    2,
+        bool, "flag", 1);
+}
+A obj;
+obj.x = 2;
+obj.z = obj.x;
+----
+
+The example above creates a bitfield pack of eight bits, which fit in
+one $(D_PARAM ubyte). The bitfields are allocated starting from the
+least significant bit, i.e. x occupies the two least significant bits
+of the bitfields storage.
+
+The sum of all bit lengths in one $(D_PARAM bitfield) instantiation
+must be exactly 8, 16, 32, or 64. If padding is needed, just allocate
+one bitfield with an empty name.
+
+Example:
+
+----
+struct A
+{
+    mixin(bitfields!(
+        bool        , "flag1",    1,
+        Flag!"flag2", "flag2",    1,
+        uint        , ""     ,    6));
+}
+----
+
+The type of a bit field can be any integral type, enumerated
+type, or type that can cast to/from an integral type. The most
+efficient type to store in bitfields is $(D_PARAM bool), followed
+by unsigned types, followed by signed types.
+*/
+mixin template bitfields2(T...)
+{
+   mixin(bitfields2Code!T);
+}
+
+/**
+Template returns the mixin code used by $(LREF bitfields2). Useful
+for debugging the generated mixin code.
+
+Example:
+
+----
+struct A
+{
+    // This will print the code generated by the bitfields2 mixin below.
+    pragma(msg, bitfields2Code!(
+        bool        , "flag1",    1,
+        Flag!"flag2", "flag2",    1,
+        uint        , ""     ,    6));
+    mixin bitfields2!(
+        bool        , "flag1",    1,
+        Flag!"flag2", "flag2",    1,
+        uint        , ""     ,    6);
+}
+----
+*/
+template bitfields2Code(T...)
+{
+    enum bitfields2Code = bitfields2CodeImpl!(true, T);
+}
+private template bitfields2CodeImpl(bool useTypeAlias, T...)
+{
+    enum bitfields2CodeImpl = bitfieldsCodeGenerator();
+
+    // NOTE: the long-verbose name for this function is helpful for compiler error messages
+    private string bitfieldsCodeGenerator()
+    {
+        string store;
+        size_t storageSize = 0;
```

Comment:
Stop creating AST nodes for something that's done automatically (i.e default initialization).